go/tasks/plugins/presto/executions_cache.go

```diff
@@ -0,0 +1,166 @@
+package presto
+
+import (
+	"context"
+	"time"
+
+	"github.com/lyft/flyteplugins/go/tasks/plugins/svc"
+
+	"k8s.io/client-go/util/workqueue"
+
+	"github.com/lyft/flytestdlib/cache"
+
+	"github.com/lyft/flyteplugins/go/tasks/errors"
+	stdErrors "github.com/lyft/flytestdlib/errors"
+
+	"github.com/lyft/flyteplugins/go/tasks/plugins/presto/client"
+	"github.com/lyft/flyteplugins/go/tasks/plugins/presto/config"
+
+	"github.com/lyft/flytestdlib/logger"
+	"github.com/lyft/flytestdlib/promutils"
+)
+
+const ResyncDuration = 3 * time.Second
+
+const (
+	BadPrestoReturnCodeError stdErrors.ErrorCode = "PRESTO_RETURNED_UNKNOWN"
+)
+
+type ExecutionsCache struct {
+	cache.AutoRefresh
+	prestoClient svc.ServiceClient
+	scope        promutils.Scope
+	cfg          *config.Config
+}
+
+func NewPrestoExecutionsCache(
+	ctx context.Context,
+	prestoClient svc.ServiceClient,
+	cfg *config.Config,
+	scope promutils.Scope) (ExecutionsCache, error) {
+
+	q := ExecutionsCache{
+		prestoClient: prestoClient,
+		scope:        scope,
+		cfg:          cfg,
+	}
+	autoRefreshCache, err := cache.NewAutoRefreshCache("presto", q.SyncPrestoQuery, workqueue.DefaultControllerRateLimiter(), ResyncDuration, cfg.Workers, cfg.LruCacheSize, scope)
```

Comment:
You should probably add rate limiter params to the config and construct a rate limiter here with those params...